create-plugin-update/action.yml

```diff
@@ -0,0 +1,85 @@
+name: "Grafana Build Plugin"
+description: "Builds a Grafana plugin"
+
+outputs:
+  archive:
+    description: "The path to the plugin archive (zip)."
+    value: ${{ steps.package-plugin.outputs.archive }}
+  archive-sha1sum:
+    description: "The path to the plugin archive sha1sum."
+    value: ${{ steps.package-plugin.outputs.archive-sha1sum }}
+
+inputs:
+  token:
+    description: "Token for the repository. Can be passed in using `{{ secrets.GITHUB_TOKEN }}`."
+    required: false
+    default: "${{ github.token }}"
+  base:
+    description: "Sets the pull request base branch"
+    required: false
+    default: "main"
+  node-version:
+    description: "Version of node"
+    required: false
+    default: "20"
+
+runs:
+  using: "composite"
+  steps:
+    - name: Setup Node.js
+      uses: actions/setup-node@v4
+      with:
+        node-version: "${{ inputs.node-version }}"
+
+    - name: Get latest version of create-plugin package
+      id: get-latest-version
+      run: echo "LATEST_VERSION=$(npx -y @grafana/create-plugin@latest version)" >> "$GITHUB_ENV"
+      shell: bash
+
+    - name: Checkout Repository
+      uses: actions/checkout@v4
+      env:
+        GITHUB_TOKEN: ${{ inputs.token }}
+
+    - name: Get version from .config/.cprc.json
+      id: get-config-version
+      run: |
+        CONFIG_VERSION=$(jq -r '.version' .config/.cprc.json)
+        echo "CONFIG_VERSION=${CONFIG_VERSION}" >> "$GITHUB_ENV"
+      shell: bash
+
+    - name: Compare versions
+      id: compare-versions
+      run: |
+        if [ "$LATEST_VERSION" != "$CONFIG_VERSION" ]; then
+          echo "update_needed=true" >> $GITHUB_OUTPUT
+        else
+          echo "update_needed=false" >> $GITHUB_OUTPUT
+        fi
+      shell: bash
+
+    - name: Update the configs
+      if: steps.compare-versions.outputs.update_needed == 'true'
+      run: |
+        ${{ github.action_path }}/pm.sh update
+        ${{ github.action_path }}/pm.sh install
+      shell: bash
+
+    - name: Create PR
+      uses: peter-evans/create-pull-request@v7
```

Comment:
if the security concern is passing the token to a thirdparty, you can always do it manually with the gh cli `gh pr create ...` ([docs](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/using-github-cli-in-workflows))

Reply:
+1 on using the github cli for this instead of a third party action.